forem/forem
# Ensure rendering parity on DisplayAd, Comment, & Article

Ready for review, but I'll add more tests.

## What type of PR is this? (check all applicable)

## Description
The `DisplayAd` rendering was similar to the `Article` and `Comment` ones, but a bit different. In this pr `DisplayAd` rendering becomes the same as `Article` and `Comments`, but preserves `DisplayAd` options.
I separated `BasicContentRenderer`(for `Comment` and `DisplayAd` - no frontmatter)  and `ContentRenderer` (with frontmatter). It's possible to have everything in one class but that would lead to have more parameters and conditions, and the code would be harder to understand.

We also have similar parsing for the Article's and Comment's previews, I decided that it's not a part of this pr, but we should implement it later (maybe when removing the feature flag?) I'm still not fan of the current implementation of `ContentRenderer`, but that is also not part of this pr.

## Related Tickets & Documents
- Related Issue #18904 
- Closes #18651

## QA Instructions, Screenshots, Recordings
- enable the `:consistent_rendering` FeatureFlag
- try to create `DisplayAd`s with liquid tags or regular display ads and check how they look 

### UI accessibility concerns?
Some of the liquid tags look weird, because they doesn't fit. Should we style them/limit liquid tag usage? Smth else?

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
